<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>New PXDataset (v2) to find and download proteomics data — PXDataset2 • rpx</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="New PXDataset (v2) to find and download proteomics data — PXDataset2" />
<meta property="og:description" content="The rpx package provides the infrastructure to access, store and
retrieve information for ProteomeXchange (PX) data sets. This can
be achieved with PXDataset2 objects can be created with the
PXDataset2() constructor that takes the unique ProteomeXchange
project identifier as input.
The new PXDataset2 class superseeds the previous and now
deprecated PXDataset version." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rpx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/rpx.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lgatto/rpx/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>New PXDataset (v2) to find and download proteomics data</h1>
    <small class="dont-index">Source: <a href='https://github.com/lgatto/rpx/blob/master/R/px2.R'><code>R/px2.R</code></a></small>
    <div class="hidden name"><code>PXDataset2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <code>rpx</code> package provides the infrastructure to access, store and
retrieve information for ProteomeXchange (PX) data sets. This can
be achieved with <code>PXDataset2</code> objects can be created with the
<code>PXDataset2()</code> constructor that takes the unique ProteomeXchange
project identifier as input.</p>
<p>The new <code>PXDataset2</code> class superseeds the previous and now
deprecated <code>PXDataset</code> version.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>PXDataset2</span><span class='op'>(</span><span class='va'>id</span>, cache <span class='op'>=</span> <span class='fu'><a href='cache.html'>rpxCache</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>PXDataset</span><span class='op'>(</span><span class='va'>id</span>, cache <span class='op'>=</span> <span class='fu'><a href='cache.html'>rpxCache</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxid</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxurl</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxtax</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxref</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='fu'>pxtitle</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxfiles</span><span class='op'>(</span><span class='va'>object</span>, n <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxCacheInfo</span><span class='op'>(</span><span class='va'>object</span><span class='op'>)</span>

<span class='co'># S4 method for PXDataset2</span>
<span class='fu'>pxget</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>list</span>, cache <span class='op'>=</span> <span class='fu'><a href='cache.html'>rpxCache</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>id</th>
      <td><p><code>character(1)</code> containing a valid ProteomeXchange
identifier.</p></td>
    </tr>
    <tr>
      <th>cache</th>
      <td><p>Object of class <code>BiocFileCache</code>. Default is to use
the central <code>rpx</code> cache returned by <code><a href='cache.html'>rpxCache()</a></code>, but users
can use their own cache. See <code><a href='cache.html'>rpxCache()</a></code> for details.</p></td>
    </tr>
    <tr>
      <th>object</th>
      <td><p>An instance of class <code>PXDataset2</code>.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p><code>integer(1)</code> indicating the number of files to be printed.</p></td>
    </tr>
    <tr>
      <th>list</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character()</a></code>, <code><a href='https://rdrr.io/r/base/numeric.html'>numeric()</a></code> or <code><a href='https://rdrr.io/r/base/logical.html'>logical()</a></code> defining the
project files to be downloaded. This list of files can
retrieved with <code>pxfiles()</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The <code>PXDataset2()</code> returns a cached <code>PXDataset2</code>
object. It thus also modifies the cache used to projet
caching, as defined by the <code>cache</code> argument.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>rpx</code> packages uses caching to store ProteomeXchange projects
and project files. When creating an object with <code>PXDataset2()</code>,
the cache is first queried for the projects identifier. If a
unique hit is found, the project is retrieved and returned. If no
matching project identifier is found, then the remote resource is
accessed to first create the new <code>PXDataset2()</code> project, then
cache it before returning it to the user. The same mechanism is
applied when project files are requested.</p>
<p>Caching is supported by BiocFileCache package. The <code>PXDataset2()</code>
constructor and the <code>px_get()</code> function can be passed a instance
of class <code>BiocFileCache</code> that defines the cache. The default is to
use the package-wide cache defined in <code><a href='cache.html'>rpxCache()</a></code>. For more
details on how to manage the cache (for example if some files need
to be deleted), please refer to the <code>BiocFileCache</code> package
vignette and documentation. See also <code><a href='cache.html'>rpxCache()</a></code> for additional
details.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<dl>
<dt><code>px_id</code></dt><dd><p><code>character(1)</code> containing the dataset's unique
ProteomeXchange identifier, as used to create the object.</p></dd>

<dt><code>px_rid</code></dt><dd><p><code>character(1)</code> storing the cached resource name in
the BiocFileCache instance stored in <code>cachepath</code>.</p></dd>

<dt><code>px_title</code></dt><dd><p><code>character(1)</code> with the project's title.</p></dd>

<dt><code>px_url</code></dt><dd><p>`character(1) with the project's URL.</p></dd>

<dt><code>px_doi</code></dt><dd><p><code>character(1)</code> with the project's DOI.</p></dd>

<dt><code>px_ref</code></dt><dd><p><code>character</code> containing the project's reference(s).</p></dd>

<dt><code>px_ref_doi</code></dt><dd><p><code>character</code> containing the project's reference DOIs.</p></dd>

<dt><code>px_pubmed</code></dt><dd><p><code>character</code> containing the project's reference
PubMed identifier.</p></dd>

<dt><code>px_files</code></dt><dd><p><code>data.frame</code> containing information about the
project files, including file names, URIs and types. The files
are retrieved from the project's README.txt file.</p></dd>

<dt><code>px_tax</code></dt><dd><p><code>charcter</code> (typically of length 1) containing the
taxonomy of the sample.</p></dd>

<dt><code>px_metadata</code></dt><dd><p><code>list</code> containing the project's metadata, as
downloaded from the ProteomeXchange site. All slots but
<code>px_files</code> are populated from this one.</p></dd>

<dt><code>cachepath</code></dt><dd><p><code>character(1)</code> storing the path to the cache the
project object is stored in.</p></dd>

</dl>
    <h2 class="hasAnchor" id="accessors"><a class="anchor" href="#accessors"></a>Accessors</h2>

    

<ul>
<li><p><code>pxfiles(object, n = 10)</code> invisibly returns all the project file
names. The function prints the first <code>n</code> files specifying
whether they are local of remote (based on the cache the object
is stored in). The printing can be ignored by wrapping the call
in <code><a href='https://rdrr.io/r/base/message.html'>suppressMessages()</a></code>.</p></li>
<li><p><code>pxget(object, list, cache)</code>: <code>list</code> is a vector defining the
files to be downloaded. If <code>list = "all"</code>, all files are
downloaded. The file names, as returned by <code>pxfiles()</code> can also
be used. Alternatively, a <code>logical</code> or <code>numeric</code> index can be
used. If missing, the file to be downloaded can be selected
from a menu.</p>
<p>The argument <code>cache</code> can be passed to define the path to the
cache. The default cache is the packages' default as returned
by <code><a href='cache.html'>rpxCache()</a></code>.</p></li>
<li><p><code>pxtax(object)</code>: returns the taxonomic name of <code>object</code>.</p></li>
<li><p><code>pxurl(object)</code>: returns the base url on the ProteomeXchange
server where the project files reside.</p></li>
<li><p><code>pxCacheInfo(object, cache): prints and invisibly returns </code>object<code>'s caching information from </code>cache<code>(default is</code>rpxCache()`). The return value is a named vector of length two
containing the resourne identifier and the cache location.</p></li>
<li><p>`pxtitle(object): returns the project's title.</p></li>
<li><p><code>pxref(object)</code>: returns the project's bibliographic
reference(s).</p></li>
</ul>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Vizcaino J.A. et al. 'ProteomeXchange: globally co-ordinated
proteomics data submission and dissemination', Nature Biotechnology 2014,
32, 223 -- 226, doi:10.1038/nbt.2839.</p>
<p>Source repository for the ProteomeXchange project:
https://code.google.com/p/proteomexchange/</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Laurent Gatto</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'></span>
<span class='r-in'><span class='va'>px</span> <span class='op'>&lt;-</span> <span class='fu'>PXDataset2</span><span class='op'>(</span><span class='st'>"PXD000001"</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading PXD000001 from cache.</span>
<span class='r-in'><span class='va'>px</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Project PXD000001 with 10 files</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Resource ID BFC1 in cache in /github/home/.cache/R/rpx.</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [1] 'erwinia_carotovora.fasta' ... [10] 'PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz'</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  Use 'pxfiles(.)' to see all files.</span>
<span class='r-in'><span class='fu'>pxtax</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "Erwinia carotovora"</span>
<span class='r-in'><span class='fu'>pxurl</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2012/03/PXD000001"</span>
<span class='r-in'><span class='fu'>pxref</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "Gatto L, Christoforou A; Using R and Bioconductor for proteomics data analysis., Biochim Biophys Acta, 2013 May 18, doi:10.1016/j.bbapap.2013.04.032 PMID:23692960"</span>
<span class='r-in'><span class='fu'>pxfiles</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Project PXD000001 files (10):</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [local]  erwinia_carotovora.fasta</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] F063721.dat</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [local]  F063721.dat-mztab.txt</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] PRIDE_Exp_Complete_Ac_22134.xml.gz</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mztab.gz</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz</span>
<span class='r-in'><span class='fu'>pxCacheInfo</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Resource ID BFC1 in cache in /github/home/.cache/R/rpx.</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>fas</span> <span class='op'>&lt;-</span> <span class='fu'>pxget</span><span class='op'>(</span><span class='va'>px</span>, <span class='st'>"erwinia_carotovora.fasta"</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading erwinia_carotovora.fasta from cache.</span>
<span class='r-in'><span class='va'>fas</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "/github/home/.cache/R/rpx/5e7462bb17c_erwinia_carotovora.fasta"</span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/Biostrings'>"Biostrings"</a></span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html'>readAAStringSet</a></span><span class='op'>(</span><span class='va'>fas</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AAStringSet object of length 4499:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>        width seq                                            names               </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [1]   147 MADITLISGSTLGSAEYVAEHL...QHQIPEDPAEEWLGSWVNLLK ECA0001 putative ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [2]   153 VAEIYQIDNLDRGILSALMENA...EIQSTETLISLQNPIMRTIAP ECA0002 AsnC-fami...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [3]   330 MKKQYIEKQQQISFVKSFFSSQ...IGQVQCGVWPQPLRESVSGLL ECA0003 putative ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [4]   492 MITLESLEMLLSIDENELLDDL...WRFDTGLKSRLMRRWQHGKAY ECA0004 conserved...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [5]   499 MRQTAALAERISRLSHALEHGL...AKIEASLQQVAEQIQQSEQQD ECA0005 conserved...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    ...   ... ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4495]   634 MSDKIIHLTDDSFDTDVLKADG...RRKVDPLRVFASDMARRLELL trx-rv3790 trx-rv...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4496]    93 MTKMNNKARRTARELKHLGASI...RELRDEFPMGYLGDYKDDDDK TimBlower TimBlower</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4497]   309 MFSNLSKRWAQRTLSKSFYSTA...KFKWAGIKTRKFVFNPPKPRK sp|P07143|CY1_YEA...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4498]   231 FPTDDDDKIVGGYTCAANSIPY...PGVYTKVCNYVNWIQQTIAAN sp|P00761|TRYP_PI...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4499]   269 GVSGSCNIDVVCPEGNGHRDVI...DAAGTGAQFIDGLDSTGTPPV sp|Q7M135|LYSC_LY...</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Laurent Gatto.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


