<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Class "PXDataset" — PXDataset-class • rpx</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Class "PXDataset" — PXDataset-class" />
<meta property="og:description" content="An S4 class to store and access information about ProteomeXchange (PX)
  data sets. Schema versions 1.0, 1.1 and 1.2 are supported and are
  documented on the PX code repository:
  https://code.google.com/p/proteomexchange/source/browse/schema/" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rpx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/rpx.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lgatto/rpx/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Class <code>"PXDataset"</code></h1>
    
    <div class="hidden name"><code>PXDataset-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>An S4 class to store and access information about ProteomeXchange (PX)
  data sets. Schema versions 1.0, 1.1 and 1.2 are supported and are
  documented on the PX code repository:
  https://code.google.com/p/proteomexchange/source/browse/schema/</p>
    </div>



    <h2 class="hasAnchor" id="objects-from-the-class"><a class="anchor" href="#objects-from-the-class"></a>Objects from the Class</h2>

    <p>Objects can be created with the constructor <code>PXDataset</code>.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    <p></p><dl>
    <dt><code>id</code>:</dt><dd><p>Object of class <code>"character"</code> storing the
      datasets unique identifier. Can be accessed with <code>pxid()</code>.</p></dd>
    <dt><code>formatVersion</code>:</dt><dd><p>Object of class <code>"character"</code>
      storing the version of the ProteomeXchange schema.</p></dd>
    <dt><code>Data</code>:</dt><dd><p>Object of class <code>"XMLNode"</code> storing the
      ProteomeXchange description as XML node tree.</p></dd>
  
</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    

<p>From version 1.99.1 on, rpx uses the Bioconductor BiocFileCache
  package to automatically cache all downloaded ProteomeXchange
  files. When a file is downloaded for the first time, it is added to
  the cache. When already available, the file path to the cached file is
  returned. The rpx chache is returned by <code>rpx:::.get_cache()</code>. The
  user is asked to confirm its creation when data is first downloaded.</p>
<p>For more details on how to manage the cache (for example if some files
  need to be deleted), please refer to the BiocFileCache vignette.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    <p></p><dl>

    <dt>pxfiles</dt><dd><p><code>signature(object = "PXDataset")</code>: return a
      <code>character</code> of all available files.</p></dd>

    <dt>pxget</dt><dd><p><code>signature(object = "PXDataset", list, ...)</code>:
      downloads the files from the ProteomeXchange repository. If
      <code>list</code> is missing, the file to be downloaded can be selected
      from a menu. If <code>list = "all"</code>, all files are downloaded. The
      file names, as returned by <code>pxfiles</code> can also be
      used. Alternatively, a <code>logical</code> or <code>numeric</code> indices
      can be used.      If not already cached, the files are downloaded and added to the
      package cache. The function then returns the names of the files in
      the cache directory.      Additional argument <code>cache</code> can be passed to the internal
      function <code>rpx:::pxget1()</code> that downloads files to and
      retrieves them from cache. The default cache is the packages'
      default <code>rappdirs::user_cache_dir(appname = "rpx")</code>.</p></dd>

    <dt>pxid</dt><dd><p><code>signature(object = "PXDataset")</code>: returns the
      unique ProteomeXchange identifier.</p></dd>

    <dt>pxref</dt><dd><p><code>signature(object = "PXDataset")</code>: returns the
      reference associcated with the <code>object</code>.</p></dd>

    <dt>pxtax</dt><dd><p><code>signature(object = "PXDataset")</code>: returns the
      scientic taxonomic name of <code>object</code>.</p></dd>

    <dt>pxurl</dt><dd><p><code>signature(object = "PXDataset"</code>: returns the base
      url on the ProteomeXchange server where <code>pxfiles(object)</code>
      reside.</p></dd>

    <dt>show</dt><dd><p><code>signature(object = "PXDataset")</code>: textual
      representation of <code>object</code>.</p></dd>
  
</dl>


    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Vizcaino J.A. et al. 'ProteomeXchange: globally co-ordinated
  proteomics data submission and dissemination', Nature Biotechnology
  2014, 32, 223 -- 226, doi:10.1038/nbt.2839.</p>
<p>Source repository for the ProteomeXchange project:
  https://code.google.com/p/proteomexchange/</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Laurent Gatto</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='va'>px</span> <span class='op'>&lt;-</span> <span class='fu'>PXDataset</span><span class='op'>(</span><span class='st'>"PXD000001"</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>px</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Object of class "PXDataset"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  Id: PXD000001 with 11 files</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [1] 'F063721.dat' ... [11] 'erwinia_carotovora.fasta'</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  Use 'pxfiles(.)' to see all files.</span>
<span class='r-in'><span class='fu'>pxtax</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "Erwinia carotovora"</span>
<span class='r-in'><span class='fu'>pxurl</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2012/03/PXD000001"</span>
<span class='r-in'><span class='fu'>pxref</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "Gatto L, Christoforou A. Using R and Bioconductor for proteomics data analysis. Biochim Biophys Acta. 2013 May 18. doi:pii: S1570-9639(13)00186-6. 10.1016/j.bbapap.2013.04.032"</span>
<span class='r-in'><span class='fu'>pxfiles</span><span class='op'>(</span><span class='va'>px</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [1] "F063721.dat"                                                         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [2] "F063721.dat-mztab.txt"                                               </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [3] "PRIDE_Exp_Complete_Ac_22134.xml.gz"                                  </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [4] "PRIDE_Exp_mzData_Ac_22134.xml.gz"                                    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [5] "PXD000001_mztab.txt"                                                 </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [6] "README.txt"                                                          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [7] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML" </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [8] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  [9] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML"         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [10] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw"           </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [11] "erwinia_carotovora.fasta"                                            </span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>fas</span> <span class='op'>&lt;-</span> <span class='fu'>pxget</span><span class='op'>(</span><span class='va'>px</span>, <span class='st'>"erwinia_carotovora.fasta"</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading erwinia_carotovora.fasta from cache.</span>
<span class='r-in'><span class='va'>fas</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "/github/home/.cache/R/rpx/e051f678efb_erwinia_carotovora.fasta"</span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/Biostrings'>"Biostrings"</a></span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: BiocGenerics</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, append,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: S4Vectors</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: stats4</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     I, expand.grid, unname</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: IRanges</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: XVector</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span>     strsplit</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html'>readAAStringSet</a></span><span class='op'>(</span><span class='va'>fas</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> AAStringSet object of length 4499:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>        width seq                                            names               </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [1]   147 MADITLISGSTLGSAEYVAEHL...QHQIPEDPAEEWLGSWVNLLK ECA0001 putative ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [2]   153 VAEIYQIDNLDRGILSALMENA...EIQSTETLISLQNPIMRTIAP ECA0002 AsnC-fami...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [3]   330 MKKQYIEKQQQISFVKSFFSSQ...IGQVQCGVWPQPLRESVSGLL ECA0003 putative ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [4]   492 MITLESLEMLLSIDENELLDDL...WRFDTGLKSRLMRRWQHGKAY ECA0004 conserved...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    [5]   499 MRQTAALAERISRLSHALEHGL...AKIEASLQQVAEQIQQSEQQD ECA0005 conserved...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    ...   ... ...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4495]   634 MSDKIIHLTDDSFDTDVLKADG...RRKVDPLRVFASDMARRLELL trx-rv3790 trx-rv...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4496]    93 MTKMNNKARRTARELKHLGASI...RELRDEFPMGYLGDYKDDDDK TimBlower TimBlower</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4497]   309 MFSNLSKRWAQRTLSKSFYSTA...KFKWAGIKTRKFVFNPPKPRK sp|P07143|CY1_YEA...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4498]   231 FPTDDDDKIVGGYTCAANSIPY...PGVYTKVCNYVNWIQQTIAAN sp|P00761|TRYP_PI...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [4499]   269 GVSGSCNIDVVCPEGNGHRDVI...DAAGTGAQFIDGLDSTGTPPV sp|Q7M135|LYSC_LY...</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Laurent Gatto.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


