<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R interface to the ProteomeXchange repository • rpx</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An R interface to the ProteomeXchange repository">
<meta property="og:description" content="rpx">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rpx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.1.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/rpx.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lgatto/rpx/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An R interface to the ProteomeXchange repository</h1>
                        <h4 data-toc-skip class="author">Laurent Gatto</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/lgatto/rpx/blob/master/vignettes/rpx.Rmd" class="external-link"><code>vignettes/rpx.Rmd</code></a></small>
      <div class="hidden name"><code>rpx.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor" aria-hidden="true"></a>Introduction</h1>
<p>The goal of the <em><a href="https://bioconductor.org/packages/3.14/rpx" class="external-link">rpx</a></em> package is to provide programmatic access to proteomics data from R, in particular to the ProteomeXchange (<a href="https://www.nature.com/articles/nbt.2839/" class="external-link">Vizcaino J.A. et al, 2014</a>) central repository (see <a href="http://www.proteomexchange.org/" class="external-link uri">http://www.proteomexchange.org/</a> and <a href="http://central.proteomexchange.org/" class="external-link uri">http://central.proteomexchange.org/</a>). Additional repositories are likely to be added in the future.</p>
</div>
<div id="the-rpx-package" class="section level1">
<h1 class="hasAnchor">
<a href="#the-rpx-package" class="anchor" aria-hidden="true"></a>The <em><a href="https://bioconductor.org/packages/3.14/rpx" class="external-link">rpx</a></em> package</h1>
<div id="pxdataset-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#pxdataset-objects" class="anchor" aria-hidden="true"></a>PXDataset objects</h2>
<p>The central object that handles data access is the <code>PXDataset</code> (version 2) class. Such an instance can be generated by passing a valid PX experiment identifier to the <code><a href="../reference/PXDataset2.html">PXDataset()</a></code> constructor.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/rpx" class="external-link">"rpx"</a></span><span class="op">)</span>
<span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"PXD000001"</span>
<span class="va">px</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PXDataset2.html">PXDataset</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading PXD000001 from cache.</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">px</span></code></pre></div>
<pre><code>## Project PXD000001 with 10 files
## </code></pre>
<pre><code>## Resource ID BFC1 in cache in /github/home/.cache/R/rpx.</code></pre>
<pre><code>##  [1] 'erwinia_carotovora.fasta' ... [10] 'PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz'
##  Use 'pxfiles(.)' to see all files.</code></pre>
</div>
<div id="data-and-meta-data" class="section level2">
<h2 class="hasAnchor">
<a href="#data-and-meta-data" class="anchor" aria-hidden="true"></a>Data and meta-data</h2>
<p>Several attributes can be extracted from an <code>PXDataset</code> projects, as described below.</p>
<p>The experiment identifier, that was originally used to create the project can be extracted with the <code><a href="../reference/PXDataset2.html">pxid()</a></code> method:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PXDataset2.html">pxid</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "PXD000001"</code></pre>
<p>The file transfer url where the data files can be accessed can be queried with the <code><a href="../reference/PXDataset2.html">pxurl()</a></code> method:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PXDataset2.html">pxurl</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2012/03/PXD000001"</code></pre>
<p>The species the data has been generated the data can be obtain calling the <code><a href="../reference/PXDataset2.html">pxtax()</a></code> function:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PXDataset2.html">pxtax</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "Erwinia carotovora"</code></pre>
<p>Relevant bibliographic references can be queried with the <code><a href="../reference/PXDataset2.html">pxref()</a></code> method:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/strwrap.html" class="external-link">strwrap</a></span><span class="op">(</span><span class="fu"><a href="../reference/PXDataset2.html">pxref</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "Gatto L, Christoforou A; Using R and Bioconductor for proteomics data"
## [2] "analysis., Biochim Biophys Acta, 2013 May 18,"                        
## [3] "doi:10.1016/j.bbapap.2013.04.032 PMID:23692960"</code></pre>
<p>All files available for the PX experiment can be obtained with the <code>pxfiles</code> method:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PXDataset2.html">pxfiles</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span></code></pre></div>
<pre><code>## Project PXD000001 files (10):
##  [local]  erwinia_carotovora.fasta
##  [remote] F063721.dat
##  [local]  F063721.dat-mztab.txt
##  [remote] PRIDE_Exp_Complete_Ac_22134.xml.gz
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML
##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mztab.gz
##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz</code></pre>
<p>The complete or partial data set can be downloaded with the <code><a href="../reference/PXDataset2.html">pxget()</a></code> function. The function takes a project instance as first mandatory argument.</p>
<p>The next argument, <code>list</code>, specifies what files to download. If missing, a menu is printed and the user can select a file. If set to <code>"all"</code>, all files of the experiment are downloaded. One of multiple file names, their indices or logicals can also be used to download specific files.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PXDataset2.html">pxget</a></span><span class="op">(</span><span class="va">px</span>, <span class="st">"F063721.dat-mztab.txt"</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading F063721.dat-mztab.txt from cache.</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span></code></pre></div>
<pre><code>## [1] "/github/home/.cache/R/rpx/c0b2b45f5ff_F063721.dat-mztab.txt"</code></pre>
<p>The <code>rpx</code> package makes use of the <em><a href="https://bioconductor.org/packages/3.14/BiocFileCache" class="external-link">BiocFileCache</a></em> package to avoid repeatedly dowloading data. When <code>PXDataset</code> projects are created and and project files are downloaded, they stored in the package’s central or a user-defined cache. Next time the project is instantiated with <code><a href="../reference/PXDataset2.html">PXDataset()</a></code> or a project file is downloaded with <code><a href="../reference/PXDataset2.html">pxget()</a></code>, existing artefacts will be retrieve from cache, instead of being created/downloaded from the remote server again. See <code><a href="../reference/cache.html">?rpxCache</a></code> for details about caching.</p>
<!-- Finally, a list of recent PX additions and updates can be obtained -->
<!-- using the `pxannounced()` function: -->
<!-- ```{r pxan} -->
<!-- pxannounced() -->
<!-- ``` -->
</div>
<div id="a-simple-use-case" class="section level2">
<h2 class="hasAnchor">
<a href="#a-simple-use-case" class="anchor" aria-hidden="true"></a>A simple use-case</h2>
<p>Below, we download the fasta file from the PXD000001 dataset and load it with the Biostrings package.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"fasta"</span>, <span class="fu"><a href="../reference/PXDataset2.html">pxfiles</a></span><span class="op">(</span><span class="va">px</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Project PXD000001 files (10):
##  [local]  erwinia_carotovora.fasta
##  [remote] F063721.dat
##  [local]  F063721.dat-mztab.txt
##  [remote] PRIDE_Exp_Complete_Ac_22134.xml.gz
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML
##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML
##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mztab.gz
##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fas</span></code></pre></div>
<pre><code>## [1] "erwinia_carotovora.fasta"</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PXDataset2.html">pxget</a></span><span class="op">(</span><span class="va">px</span>, <span class="va">fas</span><span class="op">)</span> <span class="co">## file available in the rpx cache</span></code></pre></div>
<pre><code>## Loading erwinia_carotovora.fasta from cache.</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span></code></pre></div>
<pre><code>## [1] "/github/home/.cache/R/rpx/c0b35f0beab_erwinia_carotovora.fasta"</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">"Biostrings"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></code></pre></div>
<pre><code>## AAStringSet object of length 4499:
##        width seq                                            names               
##    [1]   147 MADITLISGSTLGSAEYVAEHL...QHQIPEDPAEEWLGSWVNLLK ECA0001 putative ...
##    [2]   153 VAEIYQIDNLDRGILSALMENA...EIQSTETLISLQNPIMRTIAP ECA0002 AsnC-fami...
##    [3]   330 MKKQYIEKQQQISFVKSFFSSQ...IGQVQCGVWPQPLRESVSGLL ECA0003 putative ...
##    [4]   492 MITLESLEMLLSIDENELLDDL...WRFDTGLKSRLMRRWQHGKAY ECA0004 conserved...
##    [5]   499 MRQTAALAERISRLSHALEHGL...AKIEASLQQVAEQIQQSEQQD ECA0005 conserved...
##    ...   ... ...
## [4495]   634 MSDKIIHLTDDSFDTDVLKADG...RRKVDPLRVFASDMARRLELL trx-rv3790 trx-rv...
## [4496]    93 MTKMNNKARRTARELKHLGASI...RELRDEFPMGYLGDYKDDDDK TimBlower TimBlower
## [4497]   309 MFSNLSKRWAQRTLSKSFYSTA...KFKWAGIKTRKFVFNPPKPRK sp|P07143|CY1_YEA...
## [4498]   231 FPTDDDDKIVGGYTCAANSIPY...PGVYTKVCNYVNWIQQTIAAN sp|P00761|TRYP_PI...
## [4499]   269 GVSGSCNIDVVCPEGNGHRDVI...DAAGTGAQFIDGLDSTGTPPV sp|Q7M135|LYSC_LY...</code></pre>
</div>
</div>
<div id="questions-and-help" class="section level1">
<h1 class="hasAnchor">
<a href="#questions-and-help" class="anchor" aria-hidden="true"></a>Questions and help</h1>
<p>Either post questions on the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support forum</a> or open a GitHub <a href="https://github.com/lgatto/rpx/issues" class="external-link">issue</a>.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor" aria-hidden="true"></a>Session information</h1>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] rpx_2.1.12          Biostrings_2.61.2   GenomeInfoDb_1.29.8
## [4] XVector_0.33.0      IRanges_2.27.2      S4Vectors_0.31.5   
## [7] BiocGenerics_0.39.2 BiocStyle_2.21.3   
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.7               assertthat_0.2.1         rprojroot_2.0.2         
##  [4] digest_0.6.28            utf8_1.2.2               BiocFileCache_2.1.1     
##  [7] R6_2.5.1                 RSQLite_2.2.8            evaluate_0.14           
## [10] httr_1.4.2               pillar_1.6.3             zlibbioc_1.39.0         
## [13] rlang_0.99.0.9000        curl_4.3.2               jquerylib_0.1.4         
## [16] blob_1.2.2               rmarkdown_2.11           pkgdown_1.9000.9000.9000
## [19] textshaping_0.3.5        desc_1.4.0               stringr_1.4.0           
## [22] RCurl_1.98-1.5           bit_4.0.4                compiler_4.1.1          
## [25] xfun_0.26                pkgconfig_2.0.3          systemfonts_1.0.2       
## [28] htmltools_0.5.2          tidyselect_1.1.1         tibble_3.1.5            
## [31] GenomeInfoDbData_1.2.7   bookdown_0.24            fansi_0.5.0             
## [34] crayon_1.4.1             dplyr_1.0.7              dbplyr_2.1.1            
## [37] withr_2.4.2              bitops_1.0-7             rappdirs_0.3.3          
## [40] jsonlite_1.7.2           lifecycle_1.0.1          DBI_1.1.1               
## [43] magrittr_2.0.1           cli_3.0.1                stringi_1.7.5           
## [46] cachem_1.0.6             fs_1.5.0                 xml2_1.3.2              
## [49] bslib_0.3.0              ellipsis_0.3.2           filelock_1.0.2          
## [52] ragg_1.1.3               generics_0.1.0           vctrs_0.3.8             
## [55] tools_4.1.1              bit64_4.0.5              glue_1.4.2              
## [58] purrr_0.3.4              fastmap_1.1.0            yaml_2.2.1              
## [61] BiocManager_1.30.16      memoise_2.0.0            knitr_1.36              
## [64] sass_0.4.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
